doc/mgmt/SONiC_ConfigDB_support_Yang_default_values_and_profiles.md

```diff
@@ -0,0 +1,392 @@
+- ProfileProvider class
+  
+  - Read profile from profile DB.
+  - Merge profile to API result.
+
+- YangDefaultDecorator python class
+
+- DecoratorTable c++ class
+
+- DecoratorSubscriberStateTable c++ class
+
+## 2.4 Other code change
+
+- Add new methods to TableEntryEnumerable  interface:
+  - virtual bool hget(const std::string &key, const std::string &field, std::string &value) = 0;
+
+## 2.5 Database Schema
+
+- All profile table will have exactly same schema with existing ConfigDB tables.
+  
+  - BUFFER_POOL
+  - BUFFER_PROFILE
+
+- PROFILE_DELETE Table:
+  - For usage of this table, please refer to [Backward compatibility with existing code and applications.](#backward-compatibility-with-existing-code-and-applications)
+  - This table is a config DB table, it will presist cross reboot. When re-render config, this table will be flushed as well as other config DB tables.
+  - Yang model: [link](./sonic-profile-delete.yang "profile delete table")
+  ```
+  ; Key
+  itemkey              = 1*256VCHAR          ; Deleted profile item key.
+  ```
+
+## 2.6 Code example
+
+- Connector Decorator:
+  
+  ```
+   from swsscommon.swsscommon import SonicV2Connector, ConfigDBConnector, YangDefaultDecorator
+  
+   conn = ConfigDBConnector()
+   decorator = YangDefaultDecorator(conn)
+   decorator.connect()
+   decorator.get_table("VLAN_INTERFACE")
+   decorator.get_entry("VLAN_INTERFACE", "Vlan1000")
+   decorator.get_config()
+  ```
+
+- DecoratorTable:
+  
+  ```
+   from swsscommon.swsscommon import DBConnector, Table, DecoratorTable 
+  
+   db = DBConnector("CONFIG_DB", 0)
+   # Still can use Table to read user config:
+   # table = Table(db, 'VLAN_INTERFACE')
+   # Use DecoratorTable to read default value, profile and use config:
+   table = DecoratorTable(db, 'VLAN_INTERFACE')
+   table.get("Vlan1000")
+  ```
+
+# 3 Reboot
+
+- 
+
+## 3.1 Cold-reboot
+
+- Code-reboot will reload minigraph, profile DB will re-render in reload minigraph.
+- sonic-cfggen and sonic-utility will update to support generate profile DB during reload minigraph.
+
+## 3.2 Warm-reboot
+
+- Profile DB will follow SONiC warm-reboot process, warm-reboot will save whole Redis DB to /host/warmboot/dump.rdb, please refer to [SONiC YANG MODEL GUIDELINES](#7-1-sonic-yang-model-guidelines)
+- DB migrator will run after warm reboot, will handle OS upgrade in DB migrator.
+
+## 3.3 Fast-reboot
+
+- Fast-reboot will reload minigraph, profile DB will re-render in reload minigraph.
+- sonic-cfggen and sonic-utility will update to support generate profile DB during reload minigraph.
+
+## 3.4 Schema upgrade and DB migration
+
+- DB migrator will re-render profile DB to handle schema change.
+  - Profile DB will generate by sonic-cfggen command.
+- DB migrator code will improve by deprate hardcoded config and complex upgrade logic.
+- Some profile configuration change are disruptive, for example buffer config change,those change need fast-reboot to make sure change refelected on ASIC.
```

Comment:
IMO it should also work to do a warm reboot to upgrade the schema

Reply:
Fixed, the code/warm/fast reboot will use DB migrator to upgrade schema.